### PR TITLE
Add sequence_mask_op for DAM model

### PR DESCRIPTION
**Pull request by sneaxiy**

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -5522,3 +5523,15 @@ def flatten(x, axis=1, name=None):
         outputs={'Out': out},
         attrs={"axis": axis})
     return out
+
+
+def sequence_mask(x, max_len, mask_dtype='int64'):
```
Comment (kuke) [inline python/paddle/fluid/layers/nn.py]: Please add docs for this API, referring to http://www.paddlepaddle.org/documentation/api/zh/0.14.0/layers.html

Also please add an argument `name`

File: paddle/fluid/operators/sequence_mask_op.h [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+namespace operators {
+
+class SequenceMaskOp : public framework::OperatorWithKernel {
+ public:
+  using framework::OperatorWithKernel::OperatorWithKernel;
+
+  void InferShape(framework::InferShapeContext *ctx) const override {
+    PADDLE_ENFORCE(ctx->HasInput("X"), "Input(X) must exist");
+    auto max_len = ctx->Attrs().Get<int>("max_len");
+    PADDLE_ENFORCE_GT(max_len, 1, "Attr(max_len) must be larger than 1");
```
Comment (kuke) [inline paddle/fluid/operators/sequence_mask_op.h]: Seems not necessary

**Review by kuke — COMMENTED** [review]
Can you make a survey on what will happen when `max_len` is less than the actual maximum length of sequences.

File: python/paddle/fluid/layers/nn.py (hunk 1 above)
Comment (sneaxiy) [reply python/paddle/fluid/layers/nn.py]: Done, thanks.

File: paddle/fluid/operators/sequence_mask_op.h (hunk 2 above)
Comment (sneaxiy) [reply paddle/fluid/operators/sequence_mask_op.h]: Done, thanks.

**Review by reyoung — APPROVED** [review]

**Review by panyx0718 — APPROVED** [review]
LG API.

**Pull request merged**